crates/sui-single-node-benchmark/src/benchmark_context.rs

```diff
@@ -227,4 +227,19 @@ impl BenchmarkContext {
             *gas_objects = Arc::new(refreshed_gas_objects);
         }
     }
+
+    pub(crate) async fn validator_sign_transactions(&self, transactions: Vec<Transaction>) {
+        info!(
+            "Started signing {} transactions. You can now attach a profiler",
+            transactions.len(),
+        );
+        let tasks: FuturesUnordered<_> = transactions
+            .into_iter()
+            .map(|tx| {
+                let validator = self.validator();
+                tokio::spawn(async move { validator.sign_transaction(tx).await })
+            })
+            .collect();
+        let _results: Vec<_> = tasks.collect().await;
```

Comment:
Do we need to do anything with the result here?

Reply:
Maybe we can `unwrap()` and return the results similar to `execute_transactions_immediately()`? If we ignore the results, not sure if the compiler will optimize away everything.

Reply:
Yeah you might want to at least ensure none of them are error

Reply:
After the latest code change. The error handling is in ValidatorService. What we got is a `HandleTransactionResponse`. Is there a way to check the validity of the response?

Reply:
You can check `HandleTransactionResponse::status` and make sure it is `Signed` instead of another variant.

Reply:
Once it returns HandleTransactionResponse it's already valid so you probably could leave it alone there. The other variant is executed which shouldn't be possible here.